### ISSUE TITLE
need for speed v2

### ISSUE THREAD

**Issue by cbasavaraj** [facebookresearch/detectron2#359]
Hi,

I've trained detectron2 on a custom dataset. Training's fine and the recall / precision are pretty good. I only have a problem with the time / frame. 

## How To Reproduce the Issue
I am using 720p images. This is the relevant portion of our custom Base-RCNN-FPN config file (similar issue with faster-rcnn C4 version). Everything else including RPN and ROI_HEADS batch size are the same as in the default FPN config file.
```
  TEST: ("xxxx_2019_val",)
INPUT:
  MIN_SIZE_TRAIN: (720,)
  MAX_SIZE_TRAIN: 1280
  MIN_SIZE_TEST: 720
  MAX_SIZE_TEST: 1280
```

## Expected behavior
Model Zoo has the following numbers for R101-FPN:
```
Name | lrsched | traintime(s/iter) | inferencetime(s/im)
R101-FPN | 3x | 0.286 | 0.063
```
So that's 63 ms / image, while I get upwards of 120 ms. Appreciate any help on this, thanks!
```
chandrachud_basavaraj@instance-X:~/detectron2$ python projects/Xx/plain_train_net.py --eval-only --config-file projects/XxYy/configs/Xx-Detection/faster_rcnn_R_101_FPN_3x.yaml MODEL.WEIGHTS output/ckpts/p1/model_0007999.pth
Command Line Args: Namespace(config_file='projects/Xxyy/configs/Xx-Detection/faster_rcnn_R_101_FPN_3x.yaml', dist_url='tcp://127.0.0.1:50153', eval_only=True, machine_rank=0, num_gpus=2, num_machines=1, opts=['MODEL.WEIGHTS', 'output/ckpts/p1/model_0007999.pth'], resume=False)
....
[11/20 15:32:27 d2.evaluation.evaluator]: Inference done 1200/1294. 0.1272 s / img. ETA=0:00:11
[11/20 15:32:34 d2.evaluation.evaluator]: Inference done 1250/1294. 0.1273 s / img. ETA=0:00:05
[11/20 15:32:39 d2.evaluation.evaluator]: Total inference time: 0:02:44 (0.127230 s / img per device, on 2 devices)
[11/20 15:32:39 d2.evaluation.evaluator]: Total inference pure compute time: 0:02:38 (0.123074 s / img per device, on 2 devices)
```

## Environment
```
chandrachud_basavaraj@instance-x:~/detectron2$ python -m detectron2.utils.collect_env
------------------------  --------------------------------------------------
sys.platform              linux
Python                    3.6.8 (default, Oct  7 2019, 12:59:55) [GCC 8.3.0]
Numpy                     1.17.4
Detectron2 Compiler       GCC 7.4
Detectron2 CUDA Compiler  10.1
DETECTRON2_ENV_MODULE     <not set>
PyTorch                   1.3.0
PyTorch Debug Build       False
torchvision               0.4.1
CUDA available            True
GPU 0,1                   Tesla P100-PCIE-16GB
CUDA_HOME                 /usr/local/cuda
NVCC                      Cuda compilation tools, release 10.1, V10.1.243
Pillow                    6.0.0.post0
cv2                       4.1.0
------------------------  --------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - Intel(R) Math Kernel Library Version 2019.0.4 Product Build 20190411 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.20.5 (Git Hash 0125f28c61c1f822fd48570b4c1066f96fcb9b2e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_35,code=sm_35;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_50,code=compute_50
  - CuDNN 7.6.3
  - Magma 2.5.1
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Wno-stringop-overflow, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=True, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 
```

**Comment by ppwwyyxx** [comment]
Running the official config on P100 gives me 0.10 s/im. So your speed seems reasonable. The difference between 0.10 and 0.12 might just come from different image size or number of objects on the image.

If all your images have the same size, it might help to set `torch.backends.cudnn.benchmark = True` at the beginning of your program.

**Comment by cbasavaraj** [comment]
Thanks. I think we are spending some time on resize too as source images are 1080p. I will try without the resize, and with `torch.backends.cudnn.benchmark = True` and get back to you.

**Comment by cbasavaraj** [comment]
I just tried using a video which was already at 720p, and get mean times of 92.0 ms and 89.2 ms respectively, for without and with `torch.backends.cudnn.benchmark = True`. Finally I tested with the R101-C4 model (with cudnn benchmarking), and get 82.7 ms / frame. Like you said, the numbers are in line with yours on a Tesla P100. Like I said on the other thread though, about 10ms slower than the same model implemented here: https://github.com/jwyang/faster-rcnn.pytorch . 
Can live with this I guess, looking forward to Detectron2Go. Thanks
